core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala

```diff
@@ -172,7 +172,8 @@ private[spark] class SparkSubmitArguments(args: Seq[String], env: Map[String, St
     }
 
     // Require all python files to be local, so we can add them to the PYTHONPATH
-    if (isPython) {
+    // when yarn-cluster, all python files can be non-local
+    if (isPython && !master.equalsIgnoreCase("yarn-cluster")) {
```

Comment:
you'll probably need to `def isYarnClusterMode` or something and move the relevant code from SparkSubmit here